internal/useraccess/internal/domain/user.go

```diff
@@ -0,0 +1,37 @@
+package domain
+
+type User struct {
+	id       UserID
+	email    UserEmail
+	userName UserName
+	password HashedUserPassword
+}
+
+func NewUser(
+	userName UserName,
+	password string,
+	email UserEmail,
+) (*User, error) {
+	return &User{
+		NewUserID(),
+		email,
+		userName,
+		NewHashedUserPassword(MustNewUserPassword(password)),
+	}, nil
+}
+
+func (u User) GetID() UserID {
```

Comment:
Get rid of getters. Use snapshot to populate UserClaims. Implement method on user model to properly check user password.